lib/commands/update.js

```diff
@@ -219,10 +219,19 @@ class UpdateCommand extends Command {
 
     link({instance, installPath, version, rollback}) {
         const symlinkSync = require('symlink-or-copy').sync;
+        const semver = require('semver');
 
         fs.removeSync(path.join(process.cwd(), 'current'));
         symlinkSync(installPath, path.join(process.cwd(), 'current'));
 
+        // If upgrading from a version < 5.63.0, we need to symlink masthead so it is installed and available to update
+        if (semver.lt(instance.version, '5.63.0') && semver.gte(version, '5.63.0')) {
+            symlinkSync(
+                path.join(process.cwd(), 'current', 'content', 'themes', 'masthead'),
+                path.join(process.cwd(), 'content', 'themes', 'masthead')
+            );
+        }
+
```

Comment:
I think this should be done as a [migration](https://github.com/TryGhost/Ghost-CLI/blob/main/lib/migrations.js)

Reply:
Thanks! I tried this out because I agree it's a bit awkward to put this code here, but it didn't do what I need it to 😔. The migrations file you linked to are run depending on the _CLI version_ that [installed the instance of Ghost](https://github.com/TryGhost/Ghost-CLI/blob/a532d03231668f2076c0966760ff3c1e88f3b9e7/lib/utils/needed-migrations.js), and I need to create this symlink depending on the _Ghost instance version_. Basically, the migrations you're referring to are dependent on _how Ghost was installed_, rather than _which version of Ghost was installed_.

Basically, Ghost `5.62.0` doesn't include this theme, but `5.63.0+` will. We could of course check the Ghost version inside the migration and that would work in most cases (e.g. when upgrading sites on Ghost (Pro)), but you could theoretically use the latest version of the CLI (e.g. `1.24.4` once this change is released) to install an older version of Ghost (`5.62.0` or earlier) and then use the same version of the CLI to update to `5.63.0`. In this case, the migration wouldn't run because the CLI version hasn't changed and therefore the theme wouldn't be installed.

I'll probably break this block into its own task because I agree it doesn't really fit inside the `link` task, but I don't think using the existing migrations 'framework' works in this case.

Reply:
Ahh, yeah, the main issue I see is that older versions of Ghost are "vulnerable" to invalid symlinks (the stat call fails but isn't handled). If they supported invalid symlinks, you could create a symlink regardless of the Ghost version.

It would be great if Ghost's migrations could handle this, but you'd have to assume that everyone is using the CLI, and hardcode `current` into the path 🤮

Another option is to extend the migrations framework to also support `beforeInstance`. You would need to add rollback support, though. Then again, rollback support is required regardless.

Reply:
Okay so I spent some time refactoring this today to address some of your concerns, and I think the result is much improved from my first attempt:

- The install command no longer checks the version number — it simply creates a symlink for _all_ themes that are shipped with Ghost. So for versions < 5.67.0, it will only link `casper` because that's the only theme that ships with Ghost, for 5.67.0+ it will link `casper` and `source`, and eventually when we remove `casper` it will only install `source`.
- The update command uses basically the same logic — it will create symlinks for any theme that installs with Ghost, so when updating to 5.67.0 it will create a symlink for `source` and in the case of a rollback, it will remove any symlinks that are broken in the process of the rollback (e.g. when rolling back from 5.67.0 it will remove the symlink to `source` because it is not present in 5.66.1).
- I also added a migration to Ghost that will be included in 5.67.0 which does _nothing_ when upgrading, but when rolling back to < 5.67.0 it will check if `source` is the active_theme and update it to `casper` if so. 

These changes paired with a migration in Ghost to set the active theme to casper _only when rolling back_ to < 5.67.0 and `source` is the current theme should address all your concerns, and I think it's a more elegant solution than manually checking the version at various points in the CLI.

Reply:
Looks good overall! Left a few comments and suggestions 😊